presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -616,6 +636,24 @@ private static DateTimeField getTimestampField(ISOChronology chronology, Slice u
         throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "'" + unitString + "' is not a valid Timestamp field");
     }
 
+    private static long getUnixtimeField(Slice unit)
+    {
+        String unitString = unit.toStringUtf8().toLowerCase(ENGLISH);
+        switch (unitString) {
+            case "second":
+                return 1;
+            case "minute":
+                return 60;
+            case "hour":
+                return 3600;
+            case "day":
+                return 86400;
+            case "week":
```

Comment:
I think "quarter" and "year" are also supported

Reply:
They get complicated :( due to variable number of days

Reply:
yeah, I have the same thoughts as Kaikalur